### ISSUE TITLE
Error UIImagePNGRepresentation

### ISSUE THREAD

**Issue by pilotviper134** [fluttercandies/flutter_image_editor#15]
I am stuck on that error

Building com.test.test for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: X9AA3H6FU7
Running Xcode build...                                                  
                                                   
Xcode build done.                                           93,4s
Failed to build iOS app
Error output from Xcode build:
↳
    2020-01-10 18:32:10.216 xcodebuild[39337:1086707]  DTDeviceKit: deviceType from 10a025c153535457fc6c9f1bd506a2b199804219 was NULL
    ** BUILD FAILED **

Xcode's output:
↳
    /Users/uogforceone/documents/flutter/.pub-cache/hosted/pub.dartlang.org/image_editor-0.1.6/ios/Classes/UIImageHandler.swift:36:20: error: 'UIImagePNGRepresentation' has been replaced by instance method 'UIImage.pngData()'
                return UIImagePNGRepresentation(image)
                       ^~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~
                       image.pngData            
    UIKit.UIImagePNGRepresentation:2:13: note: 'UIImagePNGRepresentation' was obsoleted in Swift 3
    public func UIImagePNGRepresentation(_ image: UIImage) -> Data?
                ^
    /Users/uogforceone/documents/flutter/.pub-cache/hosted/pub.dartlang.org/image_editor-0.1.6/ios/Classes/UIImageHandler.swift:38:20: error: 'UIImageJPEGRepresentation' has been replaced by instance method
    'UIImage.jpegData(compressionQuality:)'
                return UIImageJPEGRepresentation(image, CGFloat(format.quality) / 100)
                       ^~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~
                       image.jpegData                   compressionQuality: 
    UIKit.UIImageJPEGRepresentation:2:13: note: 'UIImageJPEGRepresentation' was obsoleted in Swift 3
    public func UIImageJPEGRepresentation(_ image: UIImage, _ compressionQuality: CGFloat) -> Data?
                ^
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/UIKit-2LM3EQU7VVY4O.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/UIKit-2LM3EQU7VVY4O.pcm
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/Foundation-A3SOD99KJ0S9.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/Foundation-A3SOD99KJ0S9.pcm
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/MobileCoreServices-FJAFSB4GMRIG.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/MobileCoreServices-FJAFSB4GMRIG.pcm
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/ObjectiveC-1KD62J152BYGO.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/ObjectiveC-1KD62J152BYGO.pcm
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/ImageIO-1I03OOXLFD3W9.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/ImageIO-1I03OOXLFD3W9.pcm
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/CoreFoundation-10I2D2XL7L7X9.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/CoreFoundation-10I2D2XL7L7X9.pcm
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/Darwin-MI6WZSG1PNOM.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/Darwin-MI6WZSG1PNOM.pcm
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/CoreGraphics-1TH4RJJRQVFOC.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/CoreGraphics-1TH4RJJRQVFOC.pcm
    warning: /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/Photos-3GN2TRT9ZVVF.pcm: No such file or directory
    note: while processing /Users/uogforceone/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/MS5960MZXDD7/Photos-3GN2TRT9ZVVF.pcm
    note: Using new build system
    note: Planning build
    note: Constructing build description

**Comment by CaiJingLong** [comment]
The library use swift 4.0, and the method not be replaced.
What's swift version of your project ?

**Comment by CaiJingLong** [comment]
The issue of swift is so much, maybe I use Objc to rebuild the library?

**Comment by pilotviper134** [comment]
Also I use swift 4.0. I checked twice in Xcode. I will try use -ObjC flag.

**Comment by pilotviper134** [comment]
@CaiJingLong Unfortunately still the same issue.

**Comment by pilotviper134** [comment]
I think, this error telling us that on iOS 12 the old UIImageJPEGRepresentation function has been replaced with the new jpegData method on UIImage.

You need to change it, as below:
let imageData = UIImageJPEGRepresentation(image, 0.75)

to:

let imageData = image.jpegData(compressionQuality: 0.75)

**Comment by pilotviper134** [comment]
@CaiJingLong  Any update in that case?

**Comment by CaiJingLong** [comment]
As far as I know, your project swift version number does not affect the version number of the library.
I update the library's swift version to 5.0, and the example project level 4.0. It look like good.

**Comment by CaiJingLong** [comment]
![image](https://user-images.githubusercontent.com/14145407/72579814-253be480-3915-11ea-89d0-942cafd9590b.png)

![image](https://user-images.githubusercontent.com/14145407/72580132-0ab63b00-3916-11ea-89b7-a9d97f85a522.png)

![image](https://user-images.githubusercontent.com/14145407/72580158-1dc90b00-3916-11ea-822c-99df04297dc3.png)

So i don't know where is your problem.

**Comment by pilotviper134** [comment]
Unfortunetly, problem still exist 

```

Running pod install...                                              5,8s
Running Xcode build...                                                  
                                                   
Xcode build done.                                           17,8s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/uogforceone/documents/flutter/.pub-cache/hosted/pub.dartlang.org/imag
    e_editor-0.1.6/ios/Classes/UIImageHandler.swift:36:20: error:
    'UIImagePNGRepresentation' has been replaced by instance method
    'UIImage.pngData()'
                return UIImagePNGRepresentation(image)
                       ^~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~
                       image.pngData            
    UIKit.UIImagePNGRepresentation:2:13: note: 'UIImagePNGRepresentation' was
    obsoleted in Swift 3
    public func UIImagePNGRepresentation(_ image: UIImage) -> Data?
                ^
    /Users/uogforceone/documents/flutter/.pub-cache/hosted/pub.dartlang.org/imag
    e_editor-0.1.6/ios/Classes/UIImageHandler.swift:38:20: error:
    'UIImageJPEGRepresentation' has been replaced by instance method
    'UIImage.jpegData(compressionQuality:)'
                return UIImageJPEGRepresentation(image, CGFloat(format.quality)
                / 100)
                       ^~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~
                       image.jpegData                   compressionQuality: 
    UIKit.UIImageJPEGRepresentation:2:13: note: 'UIImageJPEGRepresentation' was
    obsoleted in Swift 3
    public func UIImageJPEGRepresentation(_ image: UIImage, _
    compressionQuality: CGFloat) -> Data?
                ^
    note: Using new build system
    note: Planning build
    note: Constructing build description

Encountered error while building for device.
```

**Comment by CaiJingLong** [comment]
If your code is like screenshot, the error shouldn't be report, because the UIImageJPEGRepresentation and UIImagePNGRepresentation was removed already in the code.

I level review in #16.

**Comment by pilotviper134** [comment]
Checking

**Comment by pilotviper134** [comment]
@CaiJingLong Example in review works fine :)

**Comment by CaiJingLong** [comment]
You commit and push new code to PR.
I will merge your code to master.

**Comment by pilotviper134** [comment]
You can merge code ;)

**Comment by pilotviper134** [comment]
"Only those with write access to this repository can merge pull requests." :-)

**Comment by DelattreAdixon** [comment]
Need to update the code to make it work.
```
    func outputMemory(format: FormatOption) -> Data? {
        if format.format == 0 {
            return image.pngData()
        } else {
            return image.jpegData(compressionQuality: CGFloat(format.quality) / 100)
        }
    }

```

**Comment by CaiJingLong** [comment]
To summarize, I tried using 4.0 4.1 5.0 5.1 in the main project and it can all run, so I think the problem is in your own main project configuration.

Of course, I used the final solution. In order for friends who do not understand the configuration of the iOS project to no longer have a similar situation, I completely migrated the code of the iOS part of the warehouse to the objc version.

---

And I will upgrade pub version to 0.2.0, So I close it.
